examples/camera.py

```diff
@@ -96,23 +96,71 @@ def move(self, axis_x, axis_y):
     def begin(self):
         # Set the current camera offset so you can draw your scene.
         # Translate using the zoom and the offset.
-        pyglet.gl.glTranslatef(-self.offset_x * self._zoom, -self.offset_y * self._zoom, 0)
+        pyglet.gl.glTranslatef(
+            -self.offset_x * self._zoom,
+            -self.offset_y * self._zoom, 0
+        )
 
         # Scale by zoom level.
-        pyglet.gl.glScalef(self._zoom, self._zoom, 1)
+        pyglet.gl.glScalef(
+            self._zoom,
+            self._zoom, 1
+        )
 
     def end(self):
         # Since this is a matrix, you will need to reverse the translate after rendering otherwise
         # it will multiply the current offset every draw update pushing it further and further away.
 
         # Reverse scale, since that was the last transform.
-        pyglet.gl.glScalef(1 / self._zoom, 1 / self._zoom, 1)
+        pyglet.gl.glScalef(
+            1 / self._zoom,
+            1 / self._zoom, 1
+        )
 
         # Reverse translate.
-        pyglet.gl.glTranslatef(self.offset_x * self._zoom, self.offset_y * self._zoom, 0)
+        pyglet.gl.glTranslatef(
+            self.offset_x * self._zoom,
+            self.offset_y * self._zoom, 0
+        )
 
     def __enter__(self):
         self.begin()
 
     def __exit__(self, exception_type, exception_value, traceback):
         self.end()
+
+
+class CenteredCamera(Camera):
+    """A simple 2D camera class. 0, 0 will be the centre of the screen, as opposed to the bottom left."""
+
+    def __init__(self, window: pyglet.window.Window, *args, **kwargs):
+        self.window = window
+        super().__init__(*args, **kwargs)
+
+    def begin(self):
+        x = -self.window.width//2/self._zoom + self.offset_x
```

Comment:
We may need to go with `Window.get_framebuffer_size()` here for support on HiDPI displays. 
```py
width, height = self.window.get_framebuffer_size()
....
```

Reply:
Using the framebuffer size would only really matter when setting viewport unless you intend to have 4 x larger projections for retina screens.

Reply:
Ah, you're right!

Reply:
Its probably better safe than sorry, I'll use this instead 👍

Reply:
@einarf was right and it's not needed. We're already handling that when setting the viewport, so if we used it again, we would be overcompensating.

Reply:
Gotcha, ill leave it the same then.